### ISSUE TITLE
Rails/SaveBang Fails When Used in a Conditional with an Inverse `!`.

### ISSUE THREAD

**Issue by joshuapinter** [bbatsov/rubocop#5819]
The Rails/SaveBang Cop is smart enough to know when the return value is being used in a conditional, except for when you are checking the inverse with a bang prefix. Then it raises an offense.

--------

## Expected behavior

Ignore when the return value is used in a conditional, for both normal and inverse.

i.e. I would it expect neither of the following to raise an offense.

**normal** ✅ 
```ruby
if @user.update_attributes( params[ :user ] )
```

**inverse** ✅ 
```ruby
if !@user.update_attributes( params[ :user ] )
```

## Actual behavior

Normal works fine but inverse does not.

**normal** ✅ 
```ruby
if @user.update_attributes( params[ :user ] )
```

This raises an offense:

**inverse** ❌ 
```ruby
if !@user.update_attributes( params[ :user ] )
```

## Steps to reproduce the problem

See above.

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```bash
$ bundle exec rubocop -V
0.55.0 (using Parser 2.5.1.0, running on ruby 2.3.6 x86_64-darwin17)
```
